### ISSUE TITLE
Numerical issues with float16

### ISSUE THREAD

**Issue by giorgiop** [fchollet/keras#4032]
Setting the config to `float16` alone introduces numerical issues.
The first epochs of training from the [mnist-cnn](https://github.com/fchollet/keras/blob/master/examples/imdb_cnn.py) example:

```
Epoch 1/12
60000/60000 [==============================] - 25s - loss: 0.0049 - acc: 0.0987 - val_loss: 0.0000e+00 - val_acc: 0.0980
Epoch 2/12
60000/60000 [==============================] - 24s - loss: 0.0000e+00 - acc: 0.0987 - val_loss: 0.0000e+00 - val_acc: 0.0980
Epoch 3/12
60000/60000 [==============================] - 24s - loss: 0.0000e+00 - acc: 0.0987 - val_loss: 0.0000e+00 - val_acc: 0.0980
Epoch 4/12
60000/60000 [==============================] - 24s - loss: 0.0000e+00 - acc: 0.0987 - val_loss: 0.0000e+00 - val_acc: 0.0980
Epoch 5/12
60000/60000 [==============================] - 24s - loss: 0.0000e+00 - acc: 0.0987 - val_loss: 0.0000e+00 - val_acc: 0.0980
Epoch 6/12
60000/60000 [==============================] - 24s - loss: 0.0000e+00 - acc: 0.0987 - val_loss: 0.0000e+00 - val_acc: 0.0980
```

This is with tensorflow backend.

**Comment by xsf72006** [comment]
Hi, I meet this question too. Did you solve it?

**Comment by neil-119** [comment]
I am getting the same results by switching to float16 on the NVIDIA Titan V.

Any luck?

**Comment by raghavgurbaxani** [comment]
@giorgiop  did you find a solution for the 0 loss problem mate ?

**Comment by La-Cabra** [comment]
Exact same problem with tensorflow-cpu v2.10 and the plugin for AMD GPUs. It works just fine on small datasets, but on huge datasets the loss stays permanently on 0.0000e+00. Reducing batch_size to 16 fixes it as well.
